src/OmenAgentResultMapping.sol

```diff
@@ -0,0 +1,35 @@
+// SPDX-License-Identifier: GPL-3.0
+pragma solidity ^0.8.24;
+
+import {Prediction} from "./structs.sol";
+
+/// This contract allows for mapping market addresses to IPFS hashes containing agent results.
+contract OmenAgentResultMapping {
+    event AgentResultAdded(
+        address indexed marketAddress, bytes32 indexed ipfsHash, address indexed publisherAddress, bytes32 txHash
```

Comment:
```suggestion
        address indexed marketAddress, bytes32 ipfsHash, address indexed publisherAddress, bytes32 txHash
```

Does it make sense for ipfs hash to be indexed?

Reply:
Idea of indexed arguments is to use as filters in the subgraph.
I assumed there could be interest in searching for a specific prediction - if that's not a use case, this can be removed.

Reply:
I agree - removing the ipfs Hash indexing